tests/phpunit/Differ/DiffChangedLinesParserTest.php

```diff
@@ -1,128 +0,0 @@
- * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
- * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
- * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
- * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
- * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
- * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
- */
-
-declare(strict_types=1);
-
-namespace Infection\Tests\Differ;
-
-use function array_map;
-use Generator;
-use Infection\Differ\ChangedLinesRange;
-use Infection\Differ\DiffChangedLinesParser;
-use PHPUnit\Framework\Attributes\CoversClass;
-use PHPUnit\Framework\Attributes\DataProvider;
-use PHPUnit\Framework\Attributes\Group;
-use PHPUnit\Framework\TestCase;
-use function Safe\realpath;
-
-#[Group('integration')]
-#[CoversClass(DiffChangedLinesParser::class)]
-final class DiffChangedLinesParserTest extends TestCase
-{
-    /**
-     * @param array<string, array<int, ChangedLinesRange>> $expectedMap
-     */
-    #[DataProvider('provideDiffs')]
-    public function test_it_converts_diff_to_files_and_changed_lines_map(
-        string $diff,
-        array $expectedMap,
-    ): void {
-        $collector = new DiffChangedLinesParser();
-
-        $resultMap = $collector->parse($diff);
-
-        $this->assertSame($this->convertToArray($expectedMap), $this->convertToArray($resultMap));
-    }
-
-    public static function provideDiffs(): Generator
-    {
-        yield 'one file with added lines in different places' => [
-            <<<'DIFF'
-                diff --git a/src/Container.php b/src/Container.php
-                @@ -37,0 +38 @@ namespace Infection;
-                @@ -533 +534,2 @@ final class Container
-                @@ -535,0 +538,3 @@ final class Container
-                @@ -1207,0 +1213,5 @@ final class Container
-                DIFF,
-            [
-                realpath('src/Container.php') => [
-                    new ChangedLinesRange(38, 38),
-                    new ChangedLinesRange(534, 535),
-                    new ChangedLinesRange(538, 540),
-                    new ChangedLinesRange(1213, 1217),
-                ],
-            ],
-        ];
-
-        yield 'two files, second one is new created' => [
-            <<<'DIFF'
-                diff --git a/src/Container.php b/src/Container.php
-                @@ -37,0 +38 @@ namespace Infection;
-                @@ -533 +534,2 @@ final class Container
-                @@ -535,0 +538,3 @@ final class Container
-                @@ -1207,0 +1213,5 @@ final class Container
-                diff --git a/src/Differ/FilesDiffChangedLines.php b/src/Differ/FilesDiffChangedLines.php
-                new file mode 100644
-                @@ -0,0 +1,18 @@
-                DIFF,
-            [
-                realpath('src/Container.php') => [
```

Comment:
we could use the FS to mock here instead of real paths... Those are annoying for this test (and I suppose why it's an integration test)